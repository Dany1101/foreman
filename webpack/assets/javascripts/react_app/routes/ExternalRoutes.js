import { Route } from 'react-router-dom';
import React from 'react';
import { addGlobalFill } from '../components/common/Fill/GlobalFill';
import { renderRoute } from './RoutingService';

/**
 * Adds a plugin's routes into core
 * @param  {String} id  plugin's id - can be its name
 * @param  {Array}   routes an array that contains a plugin's routes
 */
export const registerRoutes = (id, routes) =>
  routes.map(({ render, path, beforeRendering = null, ...routeProps }, index) =>
    addGlobalFill(
      'routes',
      `${id}-${index}`,
      <Route
        path={path}
        key={path}
        {...routeProps}
        render={renderProps =>
          renderRoute(render, renderProps, beforeRendering)
        }
      />
    )
  );
