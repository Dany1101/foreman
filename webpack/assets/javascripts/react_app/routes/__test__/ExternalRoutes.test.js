import { registerRoutes } from '../ExternalRoutes';
import store from '../../redux';

jest.unmock('../../redux');

const routes = [
  { path: 'path1', render: jest.fn() },
  { path: 'path12', render: jest.fn() },
];

describe('ExternalRoutes', () => {
  describe('rendering', () => {
    it('Addomg global routes', () => {
      registerRoutes('some-id', routes);
      expect(store.getState().extendable).toMatchSnapshot();
    });
  });
});
