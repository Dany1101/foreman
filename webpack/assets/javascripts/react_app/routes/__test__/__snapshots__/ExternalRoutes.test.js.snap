// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExternalRoutes rendering Addomg global routes 1`] = `
Object {
  "routes": Object {
    "some-id-0": true,
    "some-id-1": true,
  },
}
`;
